integration_test/go_java_launcher_integration_test.go

```diff
@@ -38,11 +38,35 @@ func TestMainMethod(t *testing.T) {
 	require.NoError(t, err, "failed: %s", output)
 
 	// part of expected output from launcher
-	assert.Regexp(t, `Argument list to executable binary: \[.+/bin/java -Xmx4M -Xmx1g -classpath .+/github.com/palantir/go-java-launcher/integration_test/testdata Main arg1\]`, output)
+	assert.Regexp(t, `Argument list to executable binary: \[.+/bin/java -Xmx4M -Xmx1g -classpath .+/go-java-launcher/integration_test/testdata Main arg1\]`, output)
 	// expected output of Java program
 	assert.Regexp(t, `\nmain method\n`, output)
 }
 
+func TestMainMethodJavaContainerSetsDefaults(t *testing.T) {
+	testContainerSupportEnabled(t, "testdata/launcher-custom.yml", "-XX\\:InitialRAMPercentage=80.0 -XX\\:MaxRAMPercentage=80.0")
+}
+
+func TestMainMethodJavaContainerSupportLauncherCustomInitialRamPercentageOverride(t *testing.T) {
+	testContainerSupportEnabled(t, "testdata/launcher-custom-initial-ram-percentage-override.yml", "-XX\\:InitialRAMPercentage=79.9")
+}
+
+func TestMainMethodJavaContainerSupportLauncherCustomMaxRamPercentageOverride(t *testing.T) {
+	testContainerSupportEnabled(t, "testdata/launcher-custom-max-ram-percentage-override.yml", "-XX\\:MaxRAMPercentage=79.9")
+}
+
+func TestMainMethodJavaContainerSupportLauncherCustomInitialAndMaxRamPercentageOverride(t *testing.T) {
+	testContainerSupportEnabled(t, "testdata/launcher-custom-initial-and-max-ram-percentage-override.yml", "-XX\\:InitialRAMPercentage=79.9 -XX\\:MaxRAMPercentage=80.9")
+}
```

Comment:
NOTE: This is totally fine to keep all tests as their own function, however one thing you will typically see in scenarios like this would be to use a [table-driven test](https://github.com/golang/go/wiki/TableDrivenTests), which looks something like this.
```go
func TestMainMethodJavaContainerSupport(t *testing.T) {
	for _, tc := range []struct {
		name            string
		launcherCustom  string
		expectedJVMArgs string
	}{
		{
			name:            "sets defaults",
			launcherCustom:  "testdata/launcher-custom.yml",
			expectedJVMArgs: "-XX\\:InitialRAMPercentage=80.0 -XX\\:MaxRAMPercentage=80.0",
		},
		// ... Other test cases
	} {
		t.Run(tc.name, func(t *testing.T) {
			testContainerSupportEnabled(t, tc.launcherCustom, tc.expectedJVMArgs)
		})
	}
}

```
No need to change this, but just wanted to give you some extra context.

Reply:
It's generally more idiomatic, but again both ways are fine

Reply:
I am going to switch: I quite liket he ability to just define in-method structs so effortlessly.

Reply:
Should be done.